### ISSUE TITLE
Decimal component – Prefix not correctly announced by screen readers

### ISSUE THREAD

**Issue by hagad** [Sage/carbon#7843]
### Description

There is an accessibility issue with the Decimal component when using the prefix prop.

Issue:
When navigating a Decimal input using a screen reader (VoiceOver) in Safari, the prefix (e.g., £ or $) is announced as a separate element instead of being included as part of the input’s accessible name.

Expected behavior:
Screen readers should announce adornments as part of the field label, not as independent text, i.e. "Balancing charge, £" instead of just "£"

### Reproduction

https://carbon.sage.com/?path=/docs/decimal-input--docs

### Steps to reproduce

https://github.com/user-attachments/assets/8ebb44c2-bb04-4bb1-8261-cfab2a6e3c9c

### Carbon version

158.36.0

### React version

v17

### Relevant browsers

Safari

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.
